### ISSUE TITLE
/usr/bin/ld: scripts/dtc/dtc-parser.tab.o:(.bss+0x10): multiple definition of `yylloc'; scripts/dtc/dtc-lexer.lex.o:(.bss+0x0): first defined here

### ISSUE THREAD

**Issue by muratozk** [unframework/licheepi-nano-buildroot#5]
```
From https://github.com/Lichee-Pi/u-boot
 * [new tag]               LABEL_2003_06_28_0050-stable -> LABEL_2003_06_28_0050-stable
 * [new tag]               LABEL_2003_06_28_0130-stable -> LABEL_2003_06_28_0130-stable
 * [new tag]               LABEL_2003_06_28_1800-stable -> LABEL_2003_06_28_1800-stable
 * [new tag]               U_BOOT_0_1_0                 -> U_BOOT_0_1_0
>>> uboot nano-v2018.01 Extracting
gzip -d -c /home/randa/buildroot-2020.05.1/dl/uboot/uboot-nano-v2018.01.tar.gz | /home/randa/buildroot-2020.05.1/output/host/bin/tar --strip-components=1 -C /home/randa/buildroot-2020.05.1/output/build/uboot-nano-v2018.01   -xf -
if [ -f /home/randa/buildroot-2020.05.1/output/build/uboot-nano-v2018.01/COPYING ]; then /usr/bin/install -m 0644 -D /home/randa/buildroot-2020.05.1/output/build/uboot-nano-v2018.01/COPYING /home/randa/buildroot-2020.05.1/output/build/uboot-nano-v2018.01/Licenses/gpl-2.0.txt; fi
>>> uboot nano-v2018.01 Patching
for p in  ; do if test -d $p ; then PATH=/home/randa/buildroot-2020.05.1/output/host/bin:$PATH support/scripts/apply-patches.sh  /home/randa/buildroot-2020.05.1/output/build/uboot-nano-v2018.01 $p \*.patch || exit 1 ; else PATH=/home/randa/buildroot-2020.05.1/output/host/bin:$PATH support/scripts/apply-patches.sh  /home/randa/buildroot-2020.05.1/output/build/uboot-nano-v2018.01 `dirname $p` `basename $p` || exit 1; fi done
>>> uboot nano-v2018.01 Configuring
PATH="/home/randa/buildroot-2020.05.1/output/host/bin:/home/randa/buildroot-2020.05.1/output/host/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin" AR="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-ar" AS="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-as" LD="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-ld" NM="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-nm" CC="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-gcc" GCC="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-gcc" CPP="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-cpp" CXX="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-g++" FC="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-gfortran" F77="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-gfortran" RANLIB="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-ranlib" READELF="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-readelf" STRIP="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-strip" OBJCOPY="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-objcopy" OBJDUMP="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-objdump" AR_FOR_BUILD="/usr/bin/ar" AS_FOR_BUILD="/usr/bin/as" CC_FOR_BUILD="/usr/bin/gcc" GCC_FOR_BUILD="/usr/bin/gcc" CXX_FOR_BUILD="/usr/bin/g++" LD_FOR_BUILD="/usr/bin/ld" CPPFLAGS_FOR_BUILD="-I/home/randa/buildroot-2020.05.1/output/host/include" CFLAGS_FOR_BUILD="-O2 -I/home/randa/buildroot-2020.05.1/output/host/include" CXXFLAGS_FOR_BUILD="-O2 -I/home/randa/buildroot-2020.05.1/output/host/include" LDFLAGS_FOR_BUILD="-L/home/randa/buildroot-2020.05.1/output/host/lib -Wl,-rpath,/home/randa/buildroot-2020.05.1/output/host/lib" FCFLAGS_FOR_BUILD="" DEFAULT_ASSEMBLER="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-as" DEFAULT_LINKER="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-ld" CPPFLAGS="-D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64" CFLAGS="-D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -Os -g2 " CXXFLAGS="-D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -Os -g2 " LDFLAGS="" FCFLAGS=" -Os -g2" FFLAGS=" -Os -g2" PKG_CONFIG="/home/randa/buildroot-2020.05.1/output/host/bin/pkg-config" STAGING_DIR="/home/randa/buildroot-2020.05.1/output/host/arm-buildroot-linux-gnueabi/sysroot" INTLTOOL_PERL=/usr/bin/perl /usr/bin/make  -C /home/randa/buildroot-2020.05.1/output/build/uboot-nano-v2018.01 CROSS_COMPILE="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-" ARCH=arm HOSTCC="/usr/bin/gcc -O2 -isystem /home/randa/buildroot-2020.05.1/output/host/include" HOSTLDFLAGS="-L/home/randa/buildroot-2020.05.1/output/host/lib -Wl,-rpath,/home/randa/buildroot-2020.05.1/output/host/lib"  licheepi_nano_config
make[2]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
  HOSTCC  scripts/basic/fixdep
  HOSTCC  scripts/kconfig/conf.o
  SHIPPED scripts/kconfig/zconf.tab.c
  SHIPPED scripts/kconfig/zconf.lex.c
  SHIPPED scripts/kconfig/zconf.hash.c
  HOSTCC  scripts/kconfig/zconf.tab.o
  HOSTLD  scripts/kconfig/conf
#
# configuration written to .config
#
>>> uboot nano-v2018.01 Building
PATH="/home/randa/buildroot-2020.05.1/output/host/bin:/home/randa/buildroot-2020.05.1/output/host/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin" AR="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-ar" AS="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-as" LD="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-ld" NM="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-nm" CC="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-gcc" GCC="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-gcc" CPP="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-cpp" CXX="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-g++" FC="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-gfortran" F77="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-gfortran" RANLIB="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-ranlib" READELF="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-readelf" STRIP="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-strip" OBJCOPY="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-objcopy" OBJDUMP="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-objdump" AR_FOR_BUILD="/usr/bin/ar" AS_FOR_BUILD="/usr/bin/as" CC_FOR_BUILD="/usr/bin/gcc" GCC_FOR_BUILD="/usr/bin/gcc" CXX_FOR_BUILD="/usr/bin/g++" LD_FOR_BUILD="/usr/bin/ld" CPPFLAGS_FOR_BUILD="-I/home/randa/buildroot-2020.05.1/output/host/include" CFLAGS_FOR_BUILD="-O2 -I/home/randa/buildroot-2020.05.1/output/host/include" CXXFLAGS_FOR_BUILD="-O2 -I/home/randa/buildroot-2020.05.1/output/host/include" LDFLAGS_FOR_BUILD="-L/home/randa/buildroot-2020.05.1/output/host/lib -Wl,-rpath,/home/randa/buildroot-2020.05.1/output/host/lib" FCFLAGS_FOR_BUILD="" DEFAULT_ASSEMBLER="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-as" DEFAULT_LINKER="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-ld" CPPFLAGS="-D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64" CFLAGS="-D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -Os -g2 " CXXFLAGS="-D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -Os -g2 " LDFLAGS="" FCFLAGS=" -Os -g2" FFLAGS=" -Os -g2" PKG_CONFIG="/home/randa/buildroot-2020.05.1/output/host/bin/pkg-config" STAGING_DIR="/home/randa/buildroot-2020.05.1/output/host/arm-buildroot-linux-gnueabi/sysroot" INTLTOOL_PERL=/usr/bin/perl /usr/bin/make  -C /home/randa/buildroot-2020.05.1/output/build/uboot-nano-v2018.01 CROSS_COMPILE="/home/randa/buildroot-2020.05.1/output/host/bin/arm-buildroot-linux-gnueabi-" ARCH=arm HOSTCC="/usr/bin/gcc -O2 -isystem /home/randa/buildroot-2020.05.1/output/host/include" HOSTLDFLAGS="-L/home/randa/buildroot-2020.05.1/output/host/lib -Wl,-rpath,/home/randa/buildroot-2020.05.1/output/host/lib"  all
scripts/kconfig/conf  --silentoldconfig Kconfig
  CHK     include/config.h
  UPD     include/config.h
  CFG     u-boot.cfg
  GEN     include/autoconf.mk.dep
  CFG     spl/u-boot.cfg
  GEN     include/autoconf.mk
  GEN     spl/include/autoconf.mk
  CHK     include/config/uboot.release
  CHK     include/generated/timestamp_autogenerated.h
  UPD     include/generated/timestamp_autogenerated.h
  UPD     include/config/uboot.release
  HOSTCC  scripts/dtc/dtc.o
  HOSTCC  scripts/dtc/flattree.o
  HOSTCC  scripts/dtc/fstree.o
  HOSTCC  scripts/dtc/data.o
  HOSTCC  scripts/dtc/livetree.o
  HOSTCC  scripts/dtc/treesource.o
  SHIPPED scripts/dtc/pylibfdt/libfdt.i
  PYMOD   scripts/dtc/pylibfdt/_libfdt.so
  CHK     include/generated/version_autogenerated.h
  UPD     include/generated/version_autogenerated.h
  HOSTCC  scripts/dtc/srcpos.o
  CC      lib/asm-offsets.s
  CC      arch/arm/lib/asm-offsets.s
  HOSTCC  scripts/dtc/checks.o
  HOSTCC  scripts/dtc/util.o
  CHK     include/config.h
  CHK     include/generated/generic-asm-offsets.h
  UPD     include/generated/generic-asm-offsets.h
  SHIPPED scripts/dtc/dtc-lexer.lex.c
  CFG     u-boot.cfg
  SHIPPED scripts/dtc/dtc-parser.tab.h
  SHIPPED scripts/dtc/dtc-parser.tab.c
  HOSTCC  scripts/dtc/dtc-lexer.lex.o
  HOSTCC  scripts/dtc/dtc-parser.tab.o
  CHK     include/generated/asm-offsets.h
  UPD     include/generated/asm-offsets.h
  LDS     u-boot.lds
  HOSTLD  scripts/dtc/dtc
/usr/bin/ld: scripts/dtc/dtc-parser.tab.o:(.bss+0x10): multiple definition of `yylloc'; scripts/dtc/dtc-lexer.lex.o:(.bss+0x0): first defined here
collect2: error: ld returned 1 exit status
make[4]: *** [scripts/Makefile.host:108: scripts/dtc/dtc] Error 1
make[4]: *** Waiting for unfinished jobs....
make[3]: *** [scripts/Makefile.build:425: scripts/dtc] Error 2
make[2]: *** [Makefile:491: scripts] Error 2
make[1]: *** [package/pkg-generic.mk:269: /home/randa/buildroot-2020.05.1/output/build/uboot-nano-v2018.01/.stamp_built] Error 2
make: *** [Makefile:84: _all] Error 2

```
Hi @unframework . I try it your repository. I have a one more problem. Do you have an idea ?
OS: Linux Mint 20.04
GCC Version: gcc version 10.2.0 (Ubuntu 10.2.0-5ubuntu1~20.04)
LD Version: GNU ld (GNU Binutils for Ubuntu) 2.34
Buildroot Versions: 2020.02, 2020.05.1 and 2020.02.10 test it not working..

Thank you so much!

**Comment by unframework** [comment]
Hi Murat, thanks for the report!

Unfortunately this looks like a Buildroot issue affecting a few other people too. Please see #4, and there are a few discussions online about this, such as this one: https://www.reddit.com/r/LineageOS/comments/hkq8ah/multiple_definitions_error_in_dtc_tool_when/. I am not using a newer Linux OS so I cannot reproduce and fix the bug locally yet.

Closing for now since this issue does not look related to Lichee Pi Nano; please let me know if any fixes end up applicable to this repo.

**Comment by muratozk** [comment]
Hi @unframework

I downgraded the compiler and passed that stage.
GCC Version : gcc version 9.3.0 (Ubuntu 9.3.0-17ubuntu1~20.04)

This time i ran into a different problem..
`Allocating group tables: done                            
Writing inode tables: done                            
Creating journal (1024 blocks): done
Copying files into the device: done
Writing superblocks and filesystem accounting information: done

ln -sf rootfs.ext2 /home/randa/buildroot-2020.02/output/images/rootfs.ext4
ln -snf /home/randa/buildroot-2020.02/output/host/arm-buildroot-linux-gnueabi/sysroot /home/randa/buildroot-2020.02/output/staging
>>>   Executing post-image script /home/randa/nano/board/licheepi_nano/post-image.sh
/bin/bash: /home/randa/nano/board/licheepi_nano/post-image.sh: Permission denied
make: *** [Makefile:820: target-post-image] Error 126
`

I wonder what did I do wrong ?

Thank you!

**Comment by unframework** [comment]
Looks like when you were doing the initial `git clone` of this repo the shell script somehow ended up not getting exec permissions? Can you try to run this:

```sh
chmod a+x /home/randa/nano/board/licheepi_nano/*.sh
```

... and then restart the make process (no need for full rebuild hopefully). If that fixes it I might need to add something to the docs to mention setting this exec permission.

Thanks!

**Comment by muratozk** [comment]
> Looks like when you were doing the initial `git clone` of this repo the shell script somehow ended up not getting exec permissions? Can you try to run this:
> 
> ```shell
> chmod a+x /home/randa/nano/board/licheepi_nano/*.sh
> ```
> 
> ... and then restart the make process (no need for full rebuild hopefully). If that fixes it I might need to add something to the docs to mention setting this exec permission.
> 
> Thanks!

Successful. Compiled the project. I tried serial communication and was able to connect via pins E0 and E1 (TX RX). It looks like a pretty small operating system has come up. I am changing configurations for touch and graphical application and experimenting. External tool chain did not work. In the Uboot section, it gave an error like "Your gcc is older than 6.0 and is not supported" but although I added linaro to the system variables, I could not pass that step for now. I'm experimenting with the latest version buildroot now. My main goal is to make a device where I can make a graphical voice conversation using this device. I hope I don't get any more mistakes :) Thanks.
